packages/panzer/disc-fe/src/lof/Panzer_BlockedTpetraLinearObjFactory.hpp

```diff
@@ -307,39 +343,145 @@ class BlockedTpetraLinearObjFactory : public LinearObjFactory<Traits>
    virtual void endFill(LinearObjContainer & loc) const;
 
    Teuchos::RCP<const panzer::BlockedDOFManager> getGlobalIndexer() const
-   { return blockedDOFManager_; }
+   {
+      TEUCHOS_ASSERT(!gidProviders_.empty());
+      return gidProviders_[0]->getBlockedIndexer();
+   }
 
    //! Get the domain unique global indexer this factory was created with.
    Teuchos::RCP<const panzer::GlobalIndexer> getDomainGlobalIndexer() const
-   { return blockProvider_; }
+   {
+      TEUCHOS_ASSERT(!gidProviders_.empty());
+      return gidProviders_.back()->getGlobalIndexer();
+   }
```

Comment:
I can see you're mixing `gidProviders_[0]` and `gidProviders_.back()` in this file. Is it intended? Don't really need the explanation why is that, just want to know whether it's concious decisitoon ;)

Reply:
I think we could use front() instead of [0], just to align with using back(). But it's not needed

Reply:
Yes that's concious decisitoon ;)